File: RussianSpotify.API/RussianSpotify.API.Core/Entities/Bucket.cs
namespace RussianSpotify.API.Core.Entities;

/// <summary>
/// Корзина пользователя
/// </summary>
public class Bucket
{
    /// <summary>
    /// Ид пользователя
    /// </summary>
    public Guid Id { get; set; }
    
    /// <summary>
    /// Пользователь
    /// </summary>
    public User User { get; set; }

    /// <summary>
    /// Ид пользователя
    /// </summary>
    public Guid UserId { get; set; }

    /// <summary>
    /// Песни
    /// </summary>
    public List<Song> Songs { get; set; } = new();
}

File: RussianSpotify.Admin.UI/src/models/Bucket.ts
import Song from "./Song";

export default class Bucket {
    bucketId: string;
    userId: string;
    songs: Song[] = [];

    constructor(bucketId: string,
                userId: string,
                songs: Song[]) {
        this.bucketId = bucketId;
        this.userId = userId;
        this.songs = songs;
    }
}